content/chapitres/tech-lifecycle.adoc

```diff
@@ -0,0 +1,466 @@
+** Créez un fichier `index.adoc` et remplissez-le avec le contenu donné en exemple dans la slide précédente
+** Essayez la commande de conversion dans le terminal de GitPod
+** Ouvrez le fichier HTML généré dans GitPod, et affichez l'aperçu avec l'icône en haut à droite
+
+== Exemple Make : Récapépète
+
+image::make-flow-1.svg[width=800]
+
+== Exemple Make : Introduction à Makefile
+
+* link:https://www.gnu.org/software/make/[GNU Make] est une ligne de commande,
+* qui lit un fichier `Makefile` pour exécuter des tâches.
+* Chaque tâche (ou "règle") est décrite par une "cible":
+* Format d'une "cible" make :
++
+[source,makefile]
+----
+cible: dependance
+	commandes
+----
+* On appelle la commande `make` avec une ou plusieurs cibles en argument :
++
+[source,bash]
+----
+make clean build
+----
+
+== Exemple Make : Exemple de Makefile
+
+[source,makefile]
+----
+# Fabrique le fichier "hello" (binaire) à partir des fichier "hello.o" et "main.o"
+hello: hello.o main.o
+	gcc -o hello hello.o main.o
+
+# Fabrique le fichier "hello.o" à partir du code source "hello.c"
+hello.o: hello.c
+	gcc -o hello.o -c hello.c
+
+# Fabrique le fichier "main.o" à partir du code source "main.c"
+main.o: main.c
+	gcc -o main.o -c main.c
+----
+
+[source,bash]
+----
+make hello # Appelle implicitement "make hello.o" et "make main.o"
+## équivalent à "make hello.o main.o hello"
+----
+
+== Exemple Make : Makefile - Build
+
+* *But :* on veut générer le HTML en appelant la commande `make index.html`
+* Toujours dans link:https://gitpod.io#https://github.com/cicd-lectures/demoapp[l'environnement GitPod,window=_blank],
+supprimez le fichier `index.html` si vous l'avez déjà
+* Créez un fichier `Makefile` qui contient une cible `index.html` et qui appelle la commande `asciidoctor`:
++
+[source,Makefile]
+----
+index.html:
+  asciidoctor index.adoc --out-file=$(CURDIR)/index.html
+  # $(CURDIR) est une variable Make désignant le chemin vers le dossier contenant le Makefile
+----
+
+* Essayez avec la commande `make index.html`
+
+[.notes]
+--
+=> Le fichier index.adoc doit être présent...
+--
+
+== Exercice Make : Makefile - Package
+
+* *But :* on veut générer le livrable en appelant la commande `make example-make.tar.gz`
+* On utilisera la commande `tar xzf ./example-make.tar.gz ./index.html`
+* ⚠️ Il y a une **dépendance** sur la cible `index.html` (il faut que `index.html` soit présent)
+
+=> C'est à vous !
+
+== Solution : Makefile - Package
+
+[source,Makefile]
+----
+index.html:
+  asciidoctor index.adoc --out-file=$(CURDIR)/index.html
+
+example-make.tar.gz: index.html
+  # xzf: eXtract Ze (zipped) File
+  tar xzf $(CURDIR)/example-make.tar.gz $(CURDIR)/index.html
+----
+
+[source,bash]
+----
+rm -f index.html example-make.tar.gz
+ls -l # Aucun fichier "index.html" ou "example-make.tar.gz"
+make example-make.tar.gz
+ls -l # 2 nouveaux fichiers "index.html ou "example-make.tar.gz"
+----
```

Comment:
:thinking: 

Won't the student end up in the root directory of the demoapp repo ? Which means that there will be files already present.

Reply:
Good point. The next exercise might help, but not only.
=> I'm thinking creating dedicated branches for this exercise (one empty as a playground field, and one with the working solutions) as you did for the testing. WDYT?
If ok for you, I'll do an additional PR :)

Reply:
Alright go ahead !